File: documentapi/src/vespa/documentapi/messagebus/messages/statbucketmessage.cpp
// Copyright 2017 Yahoo Holdings. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root.
#include "statbucketmessage.h"
#include "statbucketreply.h"
#include <vespa/documentapi/messagebus/documentprotocol.h>

namespace documentapi {

StatBucketMessage::StatBucketMessage() :
    DocumentMessage(),
    _bucketId(document::BucketId()),
    _documentSelection(),
    _bucketSpace("default") // TODO repo
{}

StatBucketMessage::StatBucketMessage(document::BucketId bucketId, const string& documentSelection) :
    DocumentMessage(),
    _bucketId(bucketId),
    _documentSelection(documentSelection),
    _bucketSpace("default") // TODO repo
{}

StatBucketMessage::~StatBucketMessage() {
}

DocumentReply::UP
StatBucketMessage::doCreateReply() const
{
    return DocumentReply::UP(new StatBucketReply());
}

uint32_t
StatBucketMessage::getType() const
{
    return DocumentProtocol::MESSAGE_STATBUCKET;
}

}

File: documentapi/src/main/java/com/yahoo/documentapi/messagebus/protocol/StatBucketMessage.java
// Copyright 2017 Yahoo Holdings. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root.
package com.yahoo.documentapi.messagebus.protocol;

import com.yahoo.document.BucketId;

public class StatBucketMessage extends DocumentMessage {

    private BucketId bucketId;
    private String documentSelection;

    StatBucketMessage() {
        // need to deserialize into
    }

    public StatBucketMessage(BucketId bucket, String documentSelection) {
        this.bucketId = bucket;
        this.documentSelection = documentSelection;
    }

    public BucketId getBucketId() {
        return bucketId;
    }

    void setBucketId(BucketId bucket) {
        bucketId = bucket;
    }

    public String getDocumentSelection() {
        return documentSelection;
    }

    void setDocumentSelection(String documentSelection) {
        this.documentSelection = documentSelection;
    }

    @Override
    public DocumentReply createReply() {
        return new StatBucketReply();
    }

    @Override
    public int getApproxSize() {
        return super.getApproxSize() + 8 + documentSelection.length();
    }

    @Override
    public int getType() {
        return DocumentProtocol.MESSAGE_STATBUCKET;
    }
}
